src/Microsoft.Android.Sdk.ILLink/ApplyPreserveAttributeBase.cs

```diff
@@ -138,28 +138,12 @@ void TryApplyPreserveAttribute (TypeDefinition type)
 			}
 		}
 
-		List<CustomAttribute> GetPreserveAttributes (ICustomAttributeProvider provider)
+		IEnumerable<CustomAttribute> GetPreserveAttributes (ICustomAttributeProvider provider)
 		{
-			List<CustomAttribute> attrs = new List<CustomAttribute> ();
-
-			if (!provider.HasCustomAttributes)
-				return attrs;
-
-			var attributes = provider.CustomAttributes;
-
-			for (int i = attributes.Count - 1; i >= 0; i--) {
-				var attribute = attributes [i];
-
-				bool remote_attribute;
-				if (!IsPreservedAttribute (provider, attribute, out remote_attribute))
-					continue;
-
-				attrs.Add (attribute);
-				if (remote_attribute)
-					attributes.RemoveAt (i);
-			}
-
-			return attrs;
+			yield return provider.CustomAttributes.Where (a => {
+				var type = a.Constructor.DeclaringType;
+				return type.Namespace = "Android.Runtime" && type.Name == "PreserveAttribute";
```

Comment:
This should not check the namespace, to keep the current behavior.

Reply:
I could change the but isn't this the current code https://github.com/xamarin/xamarin-android/blob/9cdaf8414614d9c9c9be79e65867ace258a33683/src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/ApplyPreserveAttribute.cs#L50-L55 ?

Reply:
There's a default branch, which uses `PreserveAttribute` in any namespace
https://github.com/xamarin/xamarin-android/blob/9cdaf8414614d9c9c9be79e65867ace258a33683/src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/ApplyPreserveAttribute.cs#L83-L87

Reply:
ok, fixed